views/Query/Collapsed.jsx

```diff
@@ -97,20 +100,28 @@ export default React.createClass({
           from: this.state.startDate.format('YYYYMMDD'),
           to: this.state.endDate.format('YYYYMMDD'),
         },
+        restrictedDateRange: this.state.restrictedDateRange,
       });
     }
   },
   dateButtonChanged(e) {
     var newDates = {};
+    var largestValue = "lasttwomonths";
+    var restrictedDateRange;
     let newButtonState = this.state.dateButtons.map((item) => {
       item.selected = item.value == e.target.value ? true : false;
       if (item.selected) {
         newDates = {startDate: item.startDate, endDate: item.endDate};
+        restrictedDateRange = item.value != largestValue;
       }
       return item;
     });
-    this.setState({dateButtons: newButtonState});
-    this.onDatesChange(newDates);
+    Promise.resolve(
+      this.setState({restrictedDateRange: restrictedDateRange})
+    ).then(() => {
```

Comment:
I would argue that this kind of sequence dependency is better handled by the React component lifecycle methods than a promise (like `componentDidUpdate`)  or that the display conditions in `views/NoResults/index.jsx` be based on state rather than props.